Classes/common/CDTFetchChanges.h

```diff
@@ -0,0 +1,163 @@
+//
+//  CDTFetchChanges.h
+//  
+//
+//  Created by Michael Rhodes on 31/03/2015.
+//
+//
+
+#import <Foundation/Foundation.h>
+
+@class CDTDatastore;
+@class CDTDocumentRevision;
+
+/**
+ Read changes to a database.
+ 
+ CDTFetchChanges reads the changes to a database from a given sequence value. It provides
+ callbacks for updated and created documents (documentChangedBlock) and deleted documents
+ (documentWithIDWasDeletedBlock). It also provides a completion block for when all changes
+ have been received and processed.
+ 
+ The CDTFetchChanges object will always return _all_ changes from the given sequence value.
+ 
+ The blocks you assign to process the fetched records are executed serially on an 
+ internal queue managed by the operation. Your blocks must be capable of executing 
+ on a background thread, so any tasks that require access to the main thread must 
+ be redirected accordingly.
+ */
+@interface CDTFetchChanges : NSObject
+
+#pragma mark Properties
+
+/**
+ The datastore whose changes should be fetched.
+ 
+ Typically this is set with the initialiser.
+ */
+@property (nonatomic, strong) CDTDatastore *datastore;
+
+//@property (nonatomic, assign) NSUInteger resultsLimit;
+//@property (nonatomic, readonly) BOOL moreComing;
+
+
+/**
+ The sequence value identifying the starting point for reading changes.
+ 
+ Each changes request returns a sequence value in the completion block. This sequence value
+ can be used to receive changes that have occured since the previous request. Treat the
+ sequence value as an opaque string; different implementations may provide differently
+ formatted values. A given sequence value should only be used with the database that
+ it was received from.
+ 
```

Comment:
The idea here is that this can be reused as an API on top of a remote changes feed eventually, to provide the same guarantees. This is also why the async API based on CloudKit is used.